src/calendar/calendar-day.tsx

```diff
@@ -169,10 +185,12 @@ export const CalendarDay = ({
             (disabledDates && disabledDates.includes(dateStartWithYear)) ||
             (currentFocus === "start" &&
                 selectedEndDate &&
-                day.isAfter(selectedEndDate)) ||
+                day.isAfter(selectedEndDate) &&
+                !isDirty) ||
             (currentFocus === "end" &&
                 selectedStartDate &&
-                day.isBefore(selectedStartDate))
+                day.isBefore(selectedStartDate) &&
+                !isDirty)
         ) {
```

Comment:
is getting too long, can improve readability by extracting this into a helper function `isDisabled`

something like

```
const isDisabled = (day): boolean => {
  const isOutsideBetweenRange = ...
  const isDisabledDate = ...
  const isStartAfterEnd = ...

  return isOutsideBetweenRange || isDisabledDate || isStartAfterEnd 
}
```